### PR TITLE
Refactor code to be more concise

### PR DESCRIPTION
**Pull request by jamesgeorge007**
An attempt made to introduce couple of refactoring, so as to have a more concise code-base.

### CODE REVIEW

**Comment by yangshun** [comment]
Thanks @jamesgeorge007, if you're going to make many refactoring PRs, we would prefer you combine them into one single PR instead of one PR per file. It's easier to review and won't unnecessarily bloat the commit history. Let us know what your thoughts are!

File: packages/docusaurus-utils/src/index.ts [hunk 1]
```diff
@@ -173,19 +173,19 @@ export function normalizeUrl(rawUrls: string[]): string {
...
+  let replaceWith = urls[0].match(/^file:\/\/\//) ? '$1:///' : '$1://';
+  urls[0] = urls[0].replace(/^([^/:]+):\/*/, replaceWith);
 
   // eslint-disable-next-line
   for (let i = 0; i < urls.length; i++) {
     let component = urls[i];
 
     if (typeof component !== 'string') {
-      throw new TypeError(`Url must be a string. Received ${component}`);
+      throw new TypeError(`Url must be a string. Received ${typeof component}`);
```
Comment (jamesgeorge007) [inline packages/docusaurus-utils/src/index.ts]: I guess this makes more sense as per the context :thinking:

**Comment by jamesgeorge007** [comment]
> Thanks @jamesgeorge007, if you're going to make many refactoring PRs, we would prefer you combine them into one single PR instead of one PR per file. It's easier to review and won't unnecessarily bloat the commit history. Let us know what your thoughts are!
> 
> P.S. Please run the prettier command before submitting or change your editor to read from the project's Prettier config.

I spotted these possibilities while going through the code-base. I shall try to submit a single PR for the future PR's of similar purpose.

**Comment by jamesgeorge007** [comment]
I'm not sure what's wrong with my local environment. Still `prettier` happens to change those files as in the previous PR :confused:

**Comment by jamesgeorge007** [comment]
@yangshun Also, I don't think this check here comes into play :thinking: 

https://github.com/facebook/docusaurus/blob/f4effe49fc5328169dd5a0aaae437a3007e349b9/packages/docusaurus/src/commands/init.ts#L64

**Comment by yangshun** [comment]
Please fix the issues and reopen after you've fixed them.

**Pull request closed without merging**